pennylane/ops/op_math/controlled_ops.py

```diff
@@ -334,3 +340,894 @@ def compute_matrix():  # pylint: disable=arguments-differ
 
     def _controlled(self, wire):
         return CCZ(wires=wire + self.wires)
+
+
+class CRX(ControlledOp):
+    r"""The controlled-RX operator
+
+    .. math::
+
+        \begin{align}
+            CR_x(\phi) &=
+            \begin{bmatrix}
+            & 1 & 0 & 0 & 0 \\
+            & 0 & 1 & 0 & 0\\
+            & 0 & 0 & \cos(\phi/2) & -i\sin(\phi/2)\\
+            & 0 & 0 & -i\sin(\phi/2) & \cos(\phi/2)
+            \end{bmatrix}.
+        \end{align}
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+    * Gradient recipe: The controlled-RX operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://doi.org/10.1088/1367-2630/ac2cb3):
+
+      .. math::
+
+          \frac{d}{d\phi}f(CR_x(\phi)) = c_+ \left[f(CR_x(\phi+a)) - f(CR_x(\phi-a))\right] - c_- \left[f(CR_x(\phi+b)) - f(CR_x(\phi-b))\right]
+
+      where :math:`f` is an expectation value depending on :math:`CR_x(\phi)`, and
+
+      - :math:`a = \pi/2`
+      - :math:`b = 3\pi/2`
+      - :math:`c_{\pm} = (\sqrt{2} \pm 1)/{4\sqrt{2}}`
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wire the operation acts on
+        id (str or None): String representing the operation (optional)
+    """
+    num_wires = 2
+    """int: Number of wires that the operation acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    name = "CRX"
+    basis = "X"
+    grad_method = "A"
+    parameter_frequencies = [(0.5, 1.0)]
+
+    def __init__(self, phi, wires, id=None):
+        super().__init__(qml.RX(phi, wires=wires[1:]), control_wires=wires[0], id=id)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        base = data[0]
+        control_wires = metadata[0]
+        return cls(*base.data, wires=control_wires + base.wires)
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        return super().label(decimals=decimals, base_label=base_label or "RX", cache=cache)
```

Comment:
I think the default `Controlled.label` should be fine here.  I don't think we need this method to be overridden on `CRX`.
```suggestion
```

Reply:
Would it change the behaviour? This was copied over from the original code for CRX.